BUG: special: test failures with CuPy 13.2.0

### Describe your issue.

I'm getting some failures locally with CuPy 13.2.0. I'm not sure whether they're specific to my machine or CuPy 13.2.0 in general, yet.

Only some of the full failure messages are included here due to the GH character limit.

### Reproducing Code Example

```python
pytest scipy/special
```


### Error message

```shell
====================================================== FAILURES =======================================================
________ test_compiles_in_cupy[ff->f-#include <special/cephes/beta.h>-out0 = special::cephes::beta(in0, in1)] _________

self = <cupy.cuda.compiler._NVRTCProgram object at 0x000001C87E8B12B0>
options = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
log_stream = None

    def compile(self, options=(), log_stream=None):
        try:
            if self.name_expressions:
                for ker in self.name_expressions:
                    nvrtc.addNameExpression(self.ptr, ker)
>           nvrtc.compileProgram(self.ptr, options)

log        = 'C:\\Users\\matth\\Desktop\\scipy\\scipy\\special\\special/cephes/../config.h(62): catastrophic error: cannot open sou...tth\\AppData\\Local\\Temp\\tmprxojmaa1\\3a1bb7f343eef2cb3e47bb9c47d2d0cdcb36ecd9.cubin.cu".\nCompilation terminated.\n'
log_stream = None
options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
self       = <cupy.cuda.compiler._NVRTCProgram object at 0x000001C87E8B12B0>

..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:677:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cupy_backends\cuda\libs\nvrtc.pyx:125: in cupy_backends.cuda.libs.nvrtc.compileProgram
    ???
cupy_backends\cuda\libs\nvrtc.pyx:138: in cupy_backends.cuda.libs.nvrtc.compileProgram
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   cupy_backends.cuda.libs.nvrtc.NVRTCError: NVRTC_ERROR_COMPILATION (6)


cupy_backends\cuda\libs\nvrtc.pyx:53: NVRTCError

During handling of the above exception, another exception occurred:

signature = 'ff->f', preamble = '#include <special/cephes/beta.h>', routine = 'out0 = special::cephes::beta(in0, in1)'
manage_cupy_cache = None

    @check_version(cupy, "13.0.0")
    @pytest.mark.parametrize("signature,preamble,routine", get_test_cases())
    @pytest.mark.xslow
    def test_compiles_in_cupy(signature, preamble, routine, manage_cupy_cache):
        name = f"x{uuid4().hex}"
        in_params, out_params = get_params(signature)

        func = cupy.ElementwiseKernel(
            in_params,
            out_params,
            routine,
            name,
            preamble=preamble,
            options=(f"--include-path={sc._get_include()}", "-std=c++17")
        )

>       _ = func(*get_sample_input(signature, cupy))

func       = <cupy._core._kernel.ElementwiseKernel object at 0x000001C870C324B0>
in_params  = 'float32 in0, float32 in1'
manage_cupy_cache = None
name       = 'x45d157b518654f99888b4aa7418127e2'
out_params = 'float32 out0'
preamble   = '#include <special/cephes/beta.h>'
routine    = 'out0 = special::cephes::beta(in0, in1)'
signature  = 'ff->f'

scipy\special\tests\test_xsf_cuda.py:110:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cupy\_core\_kernel.pyx:920: in cupy._core._kernel.ElementwiseKernel.__call__
    ???
cupy\_core\_kernel.pyx:945: in cupy._core._kernel.ElementwiseKernel._get_elementwise_kernel
    ???
cupy\_util.pyx:64: in cupy._util.memoize.decorator.ret
    ???
cupy\_core\_kernel.pyx:728: in cupy._core._kernel._get_elementwise_kernel
    ???
cupy\_core\_kernel.pyx:82: in cupy._core._kernel._get_simple_elementwise_kernel_from_code
    ???
cupy\_core\core.pyx:2258: in cupy._core.core.compile_with_cache
    ???
..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:484: in _compile_module_with_cache
    return _compile_with_cache_cuda(
        arch       = None
        backend    = 'nvrtc'
        cache_dir  = None
        cache_in_memory = False
        enable_cooperative_groups = False
        extra_source = '/*  Copyright 2008-2013 NVIDIA Corporation\n *  Copyright 2013 Filipe RNC Maia\n *\n *  Licensed under the Apache Lic...)\n#define CUDART_NAN              __longlong_as_double(0xfff8000000000000ULL)\n\n\n#endif  // CUPY_MATH_CONSTANTS_H\n'
        jitify     = False
        log_stream = None
        name_expressions = None
        options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h....get()];\nfloat &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::beta(in0, in1);\n      }\n      ;\n    }\n    '
..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:562: in _compile_with_cache_cuda
    ptx, mapping = compile_using_nvrtc(
        arch       = '75'
        backend    = 'nvrtc'
        base       = '//\n// Generated by NVIDIA NVVM Compiler\n//\n// Compiler Build ID: CL-34177558\n// Cuda compilation tools, release 12.5, V12.5.40\n// Based on NVVM 7.0.1\n//'
        cache_dir  = 'C:\\Users\\matth\\AppData\\Local\\Temp\\tmp08qw1l1y'
        cache_in_memory = False
        cu_name    = '3a1bb7f343eef2cb3e47bb9c47d2d0cdcb36ecd9.cubin.cu'
        enable_cooperative_groups = False
        env        = ('75', ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-...miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...), (12, 5), 'nvrtc', '-arch=sm_75', 'cubin')
        extra_source = '/*  Copyright 2008-2013 NVIDIA Corporation\n *  Copyright 2013 Filipe RNC Maia\n *\n *  Licensed under the Apache Lic...)\n#define CUDART_NAN              __longlong_as_double(0xfff8000000000000ULL)\n\n\n#endif  // CUPY_MATH_CONSTANTS_H\n'
        is_jitify_requested = False
        jitify     = False
        key_src    = b'(\'75\', (\'-IC:\\\\Users\\\\matth\\\\miniforge3\\\\envs\\\\scipy-dev\\\\Lib\\\\site-packages\\\\cupy\\\\_core\\\\in...)\n#define CUDART_NAN              __longlong_as_double(0xfff8000000000000ULL)\n\n\n#endif  // CUPY_MATH_CONSTANTS_H\n'
        log_stream = None
        mod        = <cupy.cuda.function.Module object at 0x000001C87D3751C0>
        name       = '3a1bb7f343eef2cb3e47bb9c47d2d0cdcb36ecd9.cubin'
        name_expressions = None
        options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
        path       = 'C:\\Users\\matth\\AppData\\Local\\Temp\\tmp08qw1l1y\\3a1bb7f343eef2cb3e47bb9c47d2d0cdcb36ecd9.cubin'
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h....get()];\nfloat &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::beta(in0, in1);\n      }\n      ;\n    }\n    '
..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:319: in compile_using_nvrtc
    return _compile(source, options, cu_path,
        _compile   = <function compile_using_nvrtc.<locals>._compile at 0x000001C87FC5CB80>
        arch       = '75'
        cache_in_memory = False
        cu_file    = <_io.TextIOWrapper name='C:\\Users\\matth\\AppData\\Local\\Temp\\tmprxojmaa1\\3a1bb7f343eef2cb3e47bb9c47d2d0cdcb36ecd9.cubin.cu' mode='w' encoding='cp1252'>
        cu_path    = 'C:\\Users\\matth\\AppData\\Local\\Temp\\tmprxojmaa1\\3a1bb7f343eef2cb3e47bb9c47d2d0cdcb36ecd9.cubin.cu'
        filename   = '3a1bb7f343eef2cb3e47bb9c47d2d0cdcb36ecd9.cubin.cu'
        jitify     = False
        log_stream = None
        name_expressions = None
        options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
        root_dir   = 'C:\\Users\\matth\\AppData\\Local\\Temp\\tmprxojmaa1'
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h....get()];\nfloat &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::beta(in0, in1);\n      }\n      ;\n    }\n    '
..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:303: in _compile
    compiled_obj, mapping = prog.compile(options, log_stream)
        arch       = '75'
        arch_opt   = '-arch=sm_75'
        cu_path    = 'C:\\Users\\matth\\AppData\\Local\\Temp\\tmprxojmaa1\\3a1bb7f343eef2cb3e47bb9c47d2d0cdcb36ecd9.cubin.cu'
        dump       = False
        headers    = ()
        include_names = ()
        jitify     = False
        log_stream = None
        major_version = 12
        method     = 'cubin'
        minor_version = 5
        name_expressions = None
        options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
        prog       = <cupy.cuda.compiler._NVRTCProgram object at 0x000001C87E8B12B0>
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h....get()];\nfloat &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::beta(in0, in1);\n      }\n      ;\n    }\n    '
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <cupy.cuda.compiler._NVRTCProgram object at 0x000001C87E8B12B0>
options = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
log_stream = None

    def compile(self, options=(), log_stream=None):
        try:
            if self.name_expressions:
                for ker in self.name_expressions:
                    nvrtc.addNameExpression(self.ptr, ker)
            nvrtc.compileProgram(self.ptr, options)
            mapping = None
            if self.name_expressions:
                mapping = {}
                for ker in self.name_expressions:
                    mapping[ker] = nvrtc.getLoweredName(self.ptr, ker)
            if log_stream is not None:
                log_stream.write(nvrtc.getProgramLog(self.ptr))
            # This is to ensure backwards compatibility with nvrtc
            if self.method == 'cubin':
                return nvrtc.getCUBIN(self.ptr), mapping
            elif self.method == 'ptx':
                return nvrtc.getPTX(self.ptr), mapping
            # TODO(leofang): support JIT LTO using nvrtc.getNVVM()?
            # need -dlto and -arch=compute_XX
            else:
                raise RuntimeError('Unknown NVRTC compile method')
        except nvrtc.NVRTCError:
            log = nvrtc.getProgramLog(self.ptr)
>           raise CompileException(log, self.src, self.name, options,
                                   'nvrtc' if not runtime.is_hip else 'hiprtc')
E           cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): catastrophic error: cannot open source file "cuda/std/cstdef"
E             #include <cuda/std/cstdef>
E                                       ^
E
E           1 catastrophic error detected in the compilation of "C:\Users\matth\AppData\Local\Temp\tmprxojmaa1\3a1bb7f343eef2cb3e47bb9c47d2d0cdcb36ecd9.cubin.cu".
E           Compilation terminated.

log        = 'C:\\Users\\matth\\Desktop\\scipy\\scipy\\special\\special/cephes/../config.h(62): catastrophic error: cannot open sou...tth\\AppData\\Local\\Temp\\tmprxojmaa1\\3a1bb7f343eef2cb3e47bb9c47d2d0cdcb36ecd9.cubin.cu".\nCompilation terminated.\n'
log_stream = None
options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
self       = <cupy.cuda.compiler._NVRTCProgram object at 0x000001C87E8B12B0>

..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:696: CompileException

...

_______ test_compiles_in_cupy[dd->d-#include <special/cephes/ellie.h>-out0 = special::cephes::ellie(in0, in1)] ________

self = <cupy.cuda.compiler._NVRTCProgram object at 0x000001C880219340>
options = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
log_stream = None

    def compile(self, options=(), log_stream=None):
        try:
            if self.name_expressions:
                for ker in self.name_expressions:
                    nvrtc.addNameExpression(self.ptr, ker)
>           nvrtc.compileProgram(self.ptr, options)

log        = 'C:\\Users\\matth\\Desktop\\scipy\\scipy\\special\\special/cephes/../config.h(62): catastrophic error: cannot open sou...tth\\AppData\\Local\\Temp\\tmp23fm_tkn\\5fd4fc262732935c0b4131a801f3e67307eb4109.cubin.cu".\nCompilation terminated.\n'
log_stream = None
options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
self       = <cupy.cuda.compiler._NVRTCProgram object at 0x000001C880219340>

..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:677:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cupy_backends\cuda\libs\nvrtc.pyx:125: in cupy_backends.cuda.libs.nvrtc.compileProgram
    ???
cupy_backends\cuda\libs\nvrtc.pyx:138: in cupy_backends.cuda.libs.nvrtc.compileProgram
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   cupy_backends.cuda.libs.nvrtc.NVRTCError: NVRTC_ERROR_COMPILATION (6)


cupy_backends\cuda\libs\nvrtc.pyx:53: NVRTCError

During handling of the above exception, another exception occurred:

signature = 'dd->d', preamble = '#include <special/cephes/ellie.h>', routine = 'out0 = special::cephes::ellie(in0, in1)'
manage_cupy_cache = None

    @check_version(cupy, "13.0.0")
    @pytest.mark.parametrize("signature,preamble,routine", get_test_cases())
    @pytest.mark.xslow
    def test_compiles_in_cupy(signature, preamble, routine, manage_cupy_cache):
        name = f"x{uuid4().hex}"
        in_params, out_params = get_params(signature)

        func = cupy.ElementwiseKernel(
            in_params,
            out_params,
            routine,
            name,
            preamble=preamble,
            options=(f"--include-path={sc._get_include()}", "-std=c++17")
        )

>       _ = func(*get_sample_input(signature, cupy))

func       = <cupy._core._kernel.ElementwiseKernel object at 0x000001C870C33320>
in_params  = 'float64 in0, float64 in1'
manage_cupy_cache = None
name       = 'xdb244347d8bb4e4bbbb4bd8ad23419b3'
out_params = 'float64 out0'
preamble   = '#include <special/cephes/ellie.h>'
routine    = 'out0 = special::cephes::ellie(in0, in1)'
signature  = 'dd->d'

scipy\special\tests\test_xsf_cuda.py:110:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cupy\_core\_kernel.pyx:920: in cupy._core._kernel.ElementwiseKernel.__call__
    ???
cupy\_core\_kernel.pyx:945: in cupy._core._kernel.ElementwiseKernel._get_elementwise_kernel
    ???
cupy\_util.pyx:64: in cupy._util.memoize.decorator.ret
    ???
cupy\_core\_kernel.pyx:728: in cupy._core._kernel._get_elementwise_kernel
    ???
cupy\_core\_kernel.pyx:82: in cupy._core._kernel._get_simple_elementwise_kernel_from_code
    ???
cupy\_core\core.pyx:2258: in cupy._core.core.compile_with_cache
    ???
..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:484: in _compile_module_with_cache
    return _compile_with_cache_cuda(
        arch       = None
        backend    = 'nvrtc'
        cache_dir  = None
        cache_in_memory = False
        enable_cooperative_groups = False
        extra_source = '/*  Copyright 2008-2013 NVIDIA Corporation\n *  Copyright 2013 Filipe RNC Maia\n *\n *  Licensed under the Apache Lic...)\n#define CUDART_NAN              __longlong_as_double(0xfff8000000000000ULL)\n\n\n#endif  // CUPY_MATH_CONSTANTS_H\n'
        jitify     = False
        log_stream = None
        name_expressions = None
        options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h...et()];\ndouble &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::ellie(in0, in1);\n      }\n      ;\n    }\n    '
..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:562: in _compile_with_cache_cuda
    ptx, mapping = compile_using_nvrtc(
        arch       = '75'
        backend    = 'nvrtc'
        base       = '//\n// Generated by NVIDIA NVVM Compiler\n//\n// Compiler Build ID: CL-34177558\n// Cuda compilation tools, release 12.5, V12.5.40\n// Based on NVVM 7.0.1\n//'
        cache_dir  = 'C:\\Users\\matth\\AppData\\Local\\Temp\\tmp08qw1l1y'
        cache_in_memory = False
        cu_name    = '5fd4fc262732935c0b4131a801f3e67307eb4109.cubin.cu'
        enable_cooperative_groups = False
        env        = ('75', ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-...miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...), (12, 5), 'nvrtc', '-arch=sm_75', 'cubin')
        extra_source = '/*  Copyright 2008-2013 NVIDIA Corporation\n *  Copyright 2013 Filipe RNC Maia\n *\n *  Licensed under the Apache Lic...)\n#define CUDART_NAN              __longlong_as_double(0xfff8000000000000ULL)\n\n\n#endif  // CUPY_MATH_CONSTANTS_H\n'
        is_jitify_requested = False
        jitify     = False
        key_src    = b'(\'75\', (\'-IC:\\\\Users\\\\matth\\\\miniforge3\\\\envs\\\\scipy-dev\\\\Lib\\\\site-packages\\\\cupy\\\\_core\\\\in...)\n#define CUDART_NAN              __longlong_as_double(0xfff8000000000000ULL)\n\n\n#endif  // CUPY_MATH_CONSTANTS_H\n'
        log_stream = None
        mod        = <cupy.cuda.function.Module object at 0x000001C87A5A3380>
        name       = '5fd4fc262732935c0b4131a801f3e67307eb4109.cubin'
        name_expressions = None
        options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
        path       = 'C:\\Users\\matth\\AppData\\Local\\Temp\\tmp08qw1l1y\\5fd4fc262732935c0b4131a801f3e67307eb4109.cubin'
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h...et()];\ndouble &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::ellie(in0, in1);\n      }\n      ;\n    }\n    '
..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:319: in compile_using_nvrtc
    return _compile(source, options, cu_path,
        _compile   = <function compile_using_nvrtc.<locals>._compile at 0x000001C87FC5DD00>
        arch       = '75'
        cache_in_memory = False
        cu_file    = <_io.TextIOWrapper name='C:\\Users\\matth\\AppData\\Local\\Temp\\tmp23fm_tkn\\5fd4fc262732935c0b4131a801f3e67307eb4109.cubin.cu' mode='w' encoding='cp1252'>
        cu_path    = 'C:\\Users\\matth\\AppData\\Local\\Temp\\tmp23fm_tkn\\5fd4fc262732935c0b4131a801f3e67307eb4109.cubin.cu'
        filename   = '5fd4fc262732935c0b4131a801f3e67307eb4109.cubin.cu'
        jitify     = False
        log_stream = None
        name_expressions = None
        options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
        root_dir   = 'C:\\Users\\matth\\AppData\\Local\\Temp\\tmp23fm_tkn'
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h...et()];\ndouble &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::ellie(in0, in1);\n      }\n      ;\n    }\n    '
..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:303: in _compile
    compiled_obj, mapping = prog.compile(options, log_stream)
        arch       = '75'
        arch_opt   = '-arch=sm_75'
        cu_path    = 'C:\\Users\\matth\\AppData\\Local\\Temp\\tmp23fm_tkn\\5fd4fc262732935c0b4131a801f3e67307eb4109.cubin.cu'
        dump       = False
        headers    = ()
        include_names = ()
        jitify     = False
        log_stream = None
        major_version = 12
        method     = 'cubin'
        minor_version = 5
        name_expressions = None
        options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
        prog       = <cupy.cuda.compiler._NVRTCProgram object at 0x000001C880219340>
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h...et()];\ndouble &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::ellie(in0, in1);\n      }\n      ;\n    }\n    '
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <cupy.cuda.compiler._NVRTCProgram object at 0x000001C880219340>
options = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
log_stream = None

    def compile(self, options=(), log_stream=None):
        try:
            if self.name_expressions:
                for ker in self.name_expressions:
                    nvrtc.addNameExpression(self.ptr, ker)
            nvrtc.compileProgram(self.ptr, options)
            mapping = None
            if self.name_expressions:
                mapping = {}
                for ker in self.name_expressions:
                    mapping[ker] = nvrtc.getLoweredName(self.ptr, ker)
            if log_stream is not None:
                log_stream.write(nvrtc.getProgramLog(self.ptr))
            # This is to ensure backwards compatibility with nvrtc
            if self.method == 'cubin':
                return nvrtc.getCUBIN(self.ptr), mapping
            elif self.method == 'ptx':
                return nvrtc.getPTX(self.ptr), mapping
            # TODO(leofang): support JIT LTO using nvrtc.getNVVM()?
            # need -dlto and -arch=compute_XX
            else:
                raise RuntimeError('Unknown NVRTC compile method')
        except nvrtc.NVRTCError:
            log = nvrtc.getProgramLog(self.ptr)
>           raise CompileException(log, self.src, self.name, options,
                                   'nvrtc' if not runtime.is_hip else 'hiprtc')
E           cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): catastrophic error: cannot open source file "cuda/std/cstdef"
E             #include <cuda/std/cstdef>
E                                       ^
E
E           1 catastrophic error detected in the compilation of "C:\Users\matth\AppData\Local\Temp\tmp23fm_tkn\5fd4fc262732935c0b4131a801f3e67307eb4109.cubin.cu".
E           Compilation terminated.

log        = 'C:\\Users\\matth\\Desktop\\scipy\\scipy\\special\\special/cephes/../config.h(62): catastrophic error: cannot open sou...tth\\AppData\\Local\\Temp\\tmp23fm_tkn\\5fd4fc262732935c0b4131a801f3e67307eb4109.cubin.cu".\nCompilation terminated.\n'
log_stream = None
options    = ('-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include/cupy/_cccl/cub', '-IC:\\Us...ecial', '-std=c++17', '-IC:\\Users\\matth\\miniforge3\\envs\\scipy-dev\\Lib\\site-packages\\cupy\\_core\\include', ...)
self       = <cupy.cuda.compiler._NVRTCProgram object at 0x000001C880219340>

..\..\miniforge3\envs\scipy-dev\Lib\site-packages\cupy\cuda\compiler.py:696: CompileException
=============================================== short test summary info ===============================================
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[ff->f-#include <special/cephes/beta.h>-out0 = special::cephes::beta(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[dd->d-#include <special/cephes/beta.h>-out0 = special::cephes::beta(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[ff->f-#include <special/binom.h>-out0 = special::binom(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/config.h(62): catastrophic ...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[dd->d-#include <special/binom.h>-out0 = special::binom(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/config.h(62): catastrophic ...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[d->d-#include <special/digamma.h>-special::digamma(in0)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[D->D-#include <special/digamma.h>-special::digamma(in0)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[f->f-#include <special/digamma.h>-special::digamma(in0)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[F->F-#include <special/digamma.h>-special::digamma(in0)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[ld->d-#include <special/cephes/expn.h>-out0 = special::cephes::expn(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[ff->f-#include <special/cephes/expn.h>-out0 = special::cephes::expn(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[dd->d-#include <special/cephes/expn.h>-out0 = special::cephes::expn(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[dddd->d-#include <special/hyp2f1.h>-out0 = special::hyp2f1(in0, in1, in2, in3)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/config.h(62): catastrophic ...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[dddD->D-#include <special/hyp2f1.h>-out0 = special::hyp2f1(in0, in1, in2, in3)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/config.h(62): catastrophic ...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[ffff->f-#include <special/hyp2f1.h>-out0 = special::hyp2f1(in0, in1, in2, in3)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/config.h(62): catastrophic ...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[fffF->F-#include <special/hyp2f1.h>-out0 = special::hyp2f1(in0, in1, in2, in3)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/config.h(62): catastrophic ...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[Dld->D-#include <special/lambertw.h>-out0 = special::lambertw(in0, in1, in2)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/config.h(62): catastrophic ...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[Flf->F-#include <special/lambertw.h>-out0 = special::lambertw(in0, in1, in2)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/config.h(62): catastrophic ...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[dd->d-#include <special/iv_ratio.h>-out0 = special::iv_ratio(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/config.h(62): catastrophic ...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[ff->f-#include <special/iv_ratio.h>-out0 = special::iv_ratio(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/config.h(62): catastrophic ...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[ff->f-#include <special/cephes/ellik.h>-out0 = special::cephes::ellik(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[dd->d-#include <special/cephes/ellik.h>-out0 = special::cephes::ellik(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[ff->f-#include <special/cephes/ellie.h>-out0 = special::cephes::ellie(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[dd->d-#include <special/cephes/ellie.h>-out0 = special::cephes::ellie(in0, in1)] - cupy.cuda.compiler.CompileException: C:\Users\matth\Desktop\scipy\scipy\special\special/cephes/../config.h(62): cat...
=================== 23 failed, 5170 passed, 249 skipped, 46 xfailed, 2 xpassed in 245.15s (0:04:05) ===================
```


### SciPy/NumPy/Python version and system information

```shell
1.15.0.dev0+1217.afeab37 2.0.0 sys.version_info(major=3, minor=12, micro=3, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: C:/Users/matth/miniforge3/envs/scipy-dev/Library/include/openblas
    lib directory: C:/Users/matth/miniforge3/envs/scipy-dev/Library/lib
    name: openblas
    openblas configuration: USE_64BITINT= NO_CBLAS= NO_LAPACK= NO_LAPACKE= DYNAMIC_ARCH=ON
      DYNAMIC_OLDER=OFF NO_AFFINITY=1 USE_OPENMP= HASWELL MAX_THREADS=128
    pc file directory: C:\Users\matth\miniforge3\envs\scipy-dev\Library\lib\pkgconfig
    version: 0.3.27
  lapack:
    detection method: pkgconfig
    found: true
    include directory: C:/Users/matth/miniforge3/envs/scipy-dev/Library/include/openblas
    lib directory: C:/Users/matth/miniforge3/envs/scipy-dev/Library/lib
    name: openblas
    openblas configuration: USE_64BITINT= NO_CBLAS= NO_LAPACK= NO_LAPACKE= DYNAMIC_ARCH=ON
      DYNAMIC_OLDER=OFF NO_AFFINITY=1 USE_OPENMP= HASWELL MAX_THREADS=128
    pc file directory: C:\Users\matth\miniforge3\envs\scipy-dev\Library\lib\pkgconfig
    version: 0.3.27
  pybind11:
    detection method: pkgconfig
    include directory: C:/Users/matth/miniforge3/envs/scipy-dev/Library/include
    name: pybind11
    version: 2.12.0
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 10.3.0
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 10.3.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 10.3.0
  pythran:
    include directory: ..\..\..\miniforge3\envs\scipy-dev\Lib\site-packages\pythran
    version: 0.16.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: windows
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: windows
Python Information:
  path: C:\Users\matth\miniforge3\envs\scipy-dev\python.exe
  version: '3.12'
```

@leofang 

`CUPY_DUMP_CUDA_SOURCE_ON_ERROR=1 SCIPY_XSLOW=1 python dev.py test -t "scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[dd->d-#include <special/cephes/ellie.h>-out0 = special::cephes::ellie(in0, in1)]"`


<details>

```
============================================================================================ FAILURES =============================================================================================
_____________________________________________ test_compiles_in_cupy[dd->d-#include <special/cephes/ellie.h>-out0 = special::cephes::ellie(in0, in1)] ______________________________________________
/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/cuda/compiler.py:677: in compile
    nvrtc.compileProgram(self.ptr, options)
        log        = '/home/treddy/github_projects/scipy/build-install/lib/python3.11/site-packages/scipy/special/special/cephes/../config....in the compilation of "/tmp/tmp3edhknim/cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin.cu".\nCompilation terminated.\n'
        log_stream = None
        options    = ('-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cccl/cub', '-I/ho...al', '-std=c++17', '-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include', ...)
        self       = <cupy.cuda.compiler._NVRTCProgram object at 0x7feb1fd0b190>
cupy_backends/cuda/libs/nvrtc.pyx:125: in cupy_backends.cuda.libs.nvrtc.compileProgram
    ???
cupy_backends/cuda/libs/nvrtc.pyx:138: in cupy_backends.cuda.libs.nvrtc.compileProgram
    ???
cupy_backends/cuda/libs/nvrtc.pyx:53: in cupy_backends.cuda.libs.nvrtc.check_status
    ???
E   cupy_backends.cuda.libs.nvrtc.NVRTCError: NVRTC_ERROR_COMPILATION (6)

During handling of the above exception, another exception occurred:
scipy/special/tests/test_xsf_cuda.py:110: in test_compiles_in_cupy
    _ = func(*get_sample_input(signature, cupy))
        func       = <cupy._core._kernel.ElementwiseKernel object at 0x7feb1f5edf30>
        in_params  = 'float64 in0, float64 in1'
        manage_cupy_cache = None
        name       = 'x7a576c0a629c4c60a1aef0e24e5b722b'
        out_params = 'float64 out0'
        preamble   = '#include <special/cephes/ellie.h>'
        routine    = 'out0 = special::cephes::ellie(in0, in1)'
        signature  = 'dd->d'
cupy/_core/_kernel.pyx:920: in cupy._core._kernel.ElementwiseKernel.__call__
    ???
cupy/_core/_kernel.pyx:945: in cupy._core._kernel.ElementwiseKernel._get_elementwise_kernel
    ???
cupy/_util.pyx:64: in cupy._util.memoize.decorator.ret
    ???
cupy/_core/_kernel.pyx:728: in cupy._core._kernel._get_elementwise_kernel
    ???
cupy/_core/_kernel.pyx:82: in cupy._core._kernel._get_simple_elementwise_kernel_from_code
    ???
cupy/_core/core.pyx:2258: in cupy._core.core.compile_with_cache
    ???
/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/cuda/compiler.py:484: in _compile_module_with_cache
    return _compile_with_cache_cuda(
        arch       = None
        backend    = 'nvrtc'
        cache_dir  = None
        cache_in_memory = False
        enable_cooperative_groups = False
        extra_source = '/*  Copyright 2008-2013 NVIDIA Corporation\n *  Copyright 2013 Filipe RNC Maia\n *\n *  Licensed under the Apache Lic...)\n#define CUDART_NAN              __longlong_as_double(0xfff8000000000000ULL)\n\n\n#endif  // CUPY_MATH_CONSTANTS_H\n'
        jitify     = False
        log_stream = None
        name_expressions = None
        options    = ('-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cccl/cub', '-I/ho...al', '-std=c++17', '-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include', ...)
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h...et()];\ndouble &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::ellie(in0, in1);\n      }\n      ;\n    }\n    '
/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/cuda/compiler.py:562: in _compile_with_cache_cuda
    ptx, mapping = compile_using_nvrtc(
        arch       = '89'
        backend    = 'nvrtc'
        base       = '//\n// Generated by NVIDIA NVVM Compiler\n//\n// Compiler Build ID: CL-32688072\n// Cuda compilation tools, release 12.1, V12.1.105\n// Based on NVVM 7.0.1\n//'
        cache_dir  = '/tmp/tmp9ux0w7gj'
        cache_in_memory = False
        cu_name    = 'cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin.cu'
        enable_cooperative_groups = False
        env        = ('89', ('-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cccl/cub',...envs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include', ...), (12, 1), 'nvrtc', '-arch=sm_89', 'cubin')
        extra_source = '/*  Copyright 2008-2013 NVIDIA Corporation\n *  Copyright 2013 Filipe RNC Maia\n *\n *  Licensed under the Apache Lic...)\n#define CUDART_NAN              __longlong_as_double(0xfff8000000000000ULL)\n\n\n#endif  // CUPY_MATH_CONSTANTS_H\n'
        is_jitify_requested = False
        jitify     = False
        key_src    = b'(\'89\', (\'-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cccl/...)\n#define CUDART_NAN              __longlong_as_double(0xfff8000000000000ULL)\n\n\n#endif  // CUPY_MATH_CONSTANTS_H\n'
        log_stream = None
        mod        = <cupy.cuda.function.Module object at 0x7feb1f14f400>
        name       = 'cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin'
        name_expressions = None
        options    = ('-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cccl/cub', '-I/ho...al', '-std=c++17', '-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include', ...)
        path       = '/tmp/tmp9ux0w7gj/cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin'
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h...et()];\ndouble &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::ellie(in0, in1);\n      }\n      ;\n    }\n    '
/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/cuda/compiler.py:319: in compile_using_nvrtc
    return _compile(source, options, cu_path,
        _compile   = <function compile_using_nvrtc.<locals>._compile at 0x7feb1f150ea0>
        arch       = '89'
        cache_in_memory = False
        cu_file    = <_io.TextIOWrapper name='/tmp/tmp3edhknim/cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin.cu' mode='w' encoding='ANSI_X3.4-1968'>
        cu_path    = '/tmp/tmp3edhknim/cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin.cu'
        filename   = 'cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin.cu'
        jitify     = False
        log_stream = None
        name_expressions = None
        options    = ('-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cccl/cub', '-I/ho...al', '-std=c++17', '-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include', ...)
        root_dir   = '/tmp/tmp3edhknim'
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h...et()];\ndouble &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::ellie(in0, in1);\n      }\n      ;\n    }\n    '
/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/cuda/compiler.py:303: in _compile
    compiled_obj, mapping = prog.compile(options, log_stream)
        arch       = '89'
        arch_opt   = '-arch=sm_89'
        cu_path    = '/tmp/tmp3edhknim/cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin.cu'
        dump       = True
        headers    = ()
        include_names = ()
        jitify     = False
        log_stream = None
        major_version = 12
        method     = 'cubin'
        minor_version = 1
        name_expressions = None
        options    = ('-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cccl/cub', '-I/ho...al', '-std=c++17', '-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include', ...)
        prog       = <cupy.cuda.compiler._NVRTCProgram object at 0x7feb1fd0b190>
        source     = '#include <cupy/complex.cuh>\n#include <cupy/carray.cuh>\n#include <cupy/atomics.cuh>\n#include <cupy/math_constants.h...et()];\ndouble &out0 = _raw_out0[_ind.get()];\nout0 = special::cephes::ellie(in0, in1);\n      }\n      ;\n    }\n    '
/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/cuda/compiler.py:696: in compile
    raise CompileException(log, self.src, self.name, options,
E   cupy.cuda.compiler.CompileException: /home/treddy/github_projects/scipy/build-install/lib/python3.11/site-packages/scipy/special/special/cephes/../config.h(62): catastrophic error: cannot open source file "cuda/std/cstdef"
E     #include <cuda/std/cstdef>
E                               ^
E   
E   1 catastrophic error detected in the compilation of "/tmp/tmp3edhknim/cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin.cu".
E   Compilation terminated.
        log        = '/home/treddy/github_projects/scipy/build-install/lib/python3.11/site-packages/scipy/special/special/cephes/../config....in the compilation of "/tmp/tmp3edhknim/cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin.cu".\nCompilation terminated.\n'
        log_stream = None
        options    = ('-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cccl/cub', '-I/ho...al', '-std=c++17', '-I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include', ...)
        self       = <cupy.cuda.compiler._NVRTCProgram object at 0x7feb1fd0b190>
-------------------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------------------
NVRTC compilation error: /home/treddy/github_projects/scipy/build-install/lib/python3.11/site-packages/scipy/special/special/cephes/../config.h(62): catastrophic error: cannot open source file "cuda/std/cstdef"
  #include <cuda/std/cstdef>
                            ^

1 catastrophic error detected in the compilation of "/tmp/tmp3edhknim/cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin.cu".
Compilation terminated.

-----
Name: /tmp/tmp3edhknim/cc8a4e9a6b61a69d00a94af9c2cdd1ecb794013a.cubin.cu
Options: -I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cccl/cub -I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cccl/thrust -I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cccl/libcudacxx --include-path=/home/treddy/github_projects/scipy/build-install/lib/python3.11/site-packages/scipy/special -std=c++17 -I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include -I/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/_core/include/cupy/_cuda/cuda-12 -ftz=true -arch=sm_89 --device-as-default-execution-space
CUDA source:
01 #include <cupy/complex.cuh>
02 #include <cupy/carray.cuh>
03 #include <cupy/atomics.cuh>
04 #include <cupy/math_constants.h>
05 
06 
07     #include <special/cephes/ellie.h>
08     extern "C" __global__ void x7a576c0a629c4c60a1aef0e24e5b722b(const CArray<double, 1, 1, 1> _raw_in0, const CArray<double, 1, 1, 1> _raw_in1, CArray<double, 1, 1, 1> _raw_out0, CIndexer<1, 1> _ind) {
09       ;
10       #pragma unroll 1
11       CUPY_FOR(i, _ind.size()) {
12         _ind.set(i);
13         const double &in0 = _raw_in0[_ind.get()];
14 const double &in1 = _raw_in1[_ind.get()];
15 double &out0 = _raw_out0[_ind.get()];
16 out0 = special::cephes::ellie(in0, in1);
17       }
18       ;
19     }
20 
-----
===================================================================================== short test summary info =====================================================================================
FAILED scipy/special/tests/test_xsf_cuda.py::test_compiles_in_cupy[dd->d-#include <special/cephes/ellie.h>-out0 = special::cephes::ellie(in0, in1)] - cupy.cuda.compiler.CompileException: /home/treddy/github_projects/scipy/build-install/lib/python3.11/site-packages/scipy/special/special/cephes/../config.h(62): catastrophic error: cannot op...
======================================================================================== 1 failed in 0.32s ========================================================================================
```

</details>

I would have installed using something similar to `pip install cupy-cuda12x`.

--- a/scipy/special/special/config.h
+++ b/scipy/special/special/config.h
@@ -59,7 +59,7 @@
 
 #include <cuda/std/cmath>
 #include <cuda/std/cstdint>
-#include <cuda/std/cstdef>
+#include <cuda/std/cstddef>
 #include <cuda/std/limits>
 #include <cuda/std/type_traits>
 #include <cuda/std/utility>

